RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1TkMu.cc

```diff
@@ -0,0 +1,413 @@
+  desc.add<double>("SetMinPtBarrelTo", 3.5);
+  desc.add<double>("SetMinPtEndcapTo", 1.0);
+  desc.add<double>("MinPL1Tk", 3.5);
+  desc.add<double>("MinPtL1TkBarrel", 3.5);
+  desc.addUntracked<bool>("UseOfflineSeed", false);
+  desc.add<bool>("UseUnassociatedL1", true);
+  desc.add<std::vector<double>>("MatchDR", {0.3});
+  desc.add<std::vector<double>>("EtaMatchingBins", {0., 2.5});
+  desc.addUntracked<edm::InputTag>("OfflineSeedLabel", edm::InputTag(""));
+
+  edm::ParameterSetDescription psd0;
+  psd0.addUntracked<std::vector<std::string>>("Propagators", {"SteppingHelixPropagatorAny"});
+  psd0.add<bool>("RPCLayers", true);
+  psd0.addUntracked<bool>("UseMuonNavigation", true);
+  desc.add<edm::ParameterSetDescription>("ServiceParameters", psd0);
+  descriptions.add("L2MuonSeedGeneratorFromL1TkMu", desc);
+}
+
+void L2MuonSeedGeneratorFromL1TkMu::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  const std::string metname = "Muon|RecoMuon|L2MuonSeedGeneratorFromL1TkMu";
+  MuonPatternRecoDumper debug;
+
+  auto output = std::make_unique<L2MuonTrajectorySeedCollection>();
+
+  edm::Handle<l1t::TkMuonCollection> muColl;
+  iEvent.getByToken(muCollToken_, muColl);
+  LogDebug(metname) << "Number of muons " << muColl->size() << endl;
+
+  edm::Handle<edm::View<TrajectorySeed>> offlineSeedHandle;
+  vector<int> offlineSeedMap;
+  if (useOfflineSeed) {
+    iEvent.getByToken(offlineSeedToken_, offlineSeedHandle);
+    offlineSeedMap = vector<int>(offlineSeedHandle->size(), 0);
+  }
+
+  for (auto ittkmu = muColl->begin(); ittkmu != muColl->end(); ittkmu++) {
+    // L1 tracker track
+    auto it = ittkmu->trkPtr();
+
+    // propagate to GMT
+    auto p3 = it->momentum();
+    float tk_pt = p3.perp();
+    float tk_p = p3.mag();
+    float tk_eta = p3.eta();
+    float tk_aeta = std::abs(tk_eta);
+    float tk_phi = p3.phi();
+    float tk_q = it->rInv() > 0 ? 1. : -1.;
+    float tk_z = it->POCA().z();
+
+    if (tk_p < theMinPL1Tk)
+      continue;
+    if (tk_aeta < 1.1 && tk_pt < theMinPtL1TkBarrel)
+      continue;
+
+    float dzCorrPhi = 1.;
+    float deta = 0;
+    float etaProp = tk_aeta;
+
+    if (tk_aeta < 1.1) {
+      etaProp = 1.1;
+      deta = tk_z / 550. / cosh(tk_aeta);
+    } else {
+      float delta = tk_z / 850.;  //roughly scales as distance to 2nd station
+      if (tk_eta > 0)
+        delta *= -1;
+      dzCorrPhi = 1. + delta;
+
+      float zOzs = tk_z / 850.;
+      if (tk_eta > 0)
+        deta = zOzs / (1. - zOzs);
+      else
+        deta = zOzs / (1. + zOzs);
+      deta = deta * tanh(tk_eta);
+    }
+    float resPhi = tk_phi - 1.464 * tk_q * cosh(1.7) / cosh(etaProp) / tk_pt * dzCorrPhi - M_PI / 144.;
+    resPhi = reco::reduceRange(resPhi);
+
+    float pt = tk_pt;  //not corrected for eloss
+    float eta = tk_eta + deta;
+    float theta = 2 * atan(exp(-eta));
+    float phi = resPhi;
+    int charge = it->rInv() > 0 ? 1 : -1;
+
+    bool barrel = tk_aeta < 1.1 ? true : false;
```

Comment:
```suggestion
    bool barrel = tk_aeta < 1.1;
```
And you can define it at the beginning, thus avoiding the several previous checks `tk_aeta < 1.1`